elitzur-core/src/main/scala/com/spotify/elitzur/validators/DynamicRecordValidator.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.elitzur.validators
+
+import com.spotify.elitzur.MetricsReporter
+
+
+@SuppressWarnings(Array("org.wartremover.warts.Var"))
+final private[elitzur] case class DynamicRecordValidator(validators: Array[Validator[Any]],
+                                                         labels: Array[String])
```

Comment:
Is the plan for nested records to flatten beforehand and have the full field name as the label?

Reply:
I guess that would hit issues for repeated-nested records actually so im assuming they are still nested here

Reply:
Yea, I think so, but wanted to leave it up to whoever implements the other part. Could be just implicitly deriving a validator for a Map-like type and using that similar to case class diffy. There is an implicit coupling here of Validators, Labels, and Values. Which is why I previously had the the `ValidatorAccessor`, but this created a really messy type setup for `DynamicRecordValidator` when I tried to use it here. Plus I think it's more analogous to how it works with the derived one having it separate, but there's probably a counterpoint to be made here.